src/plots/plots.js

```diff
@@ -559,6 +562,41 @@ function relinkPrivateKeys(toContainer, fromContainer) {
     }
 }
 
+plots.linkSubplots = function(newFullData, newFullLayout, oldFullData, oldFullLayout) {
+    var oldSubplots = oldFullLayout._plots || {},
+        newSubplots = newFullLayout._plots = {};
+
+    var mockGd = {
+        data: newFullData,
+        _fullLayout: newFullLayout
+    };
+
+    var ids = Plotly.Axes.getSubplots(mockGd);
+
+    function getAxisFunc(subplot, axLetter) {
+        return function() { return Plotly.Axes.getFromId(mockGd, subplot, axLetter); };
+    }
+
+    for(var i = 0; i < ids.length; i++) {
+        var id = ids[i],
+            oldSubplot = oldSubplots[id],
+            plotinfo;
+
+        if(oldSubplot) {
+            plotinfo = newSubplots[id] = oldSubplot;
+        }
+        else {
+            plotinfo = newSubplots[id] = {};
+            plotinfo.id = id;
+        }
+
+        plotinfo.x = getAxisFunc(id, 'x');
+        plotinfo.y = getAxisFunc(id, 'y');
```

Comment:
Here they are for backward-compatibility.

It might be worth the time to remove all `x()` and `y()` and replace them with `_plots[''].xaxis` and `_plots[''].yaxis` respectively.